pwiz_tools/Shared/ProteomeDb/API/ProteinMetadata.cs

```diff
@@ -306,7 +306,26 @@ public class WebSearchTerm: IEquatable<WebSearchTerm>
         public WebSearchTerm(char service, string query)
         {
             Service = service;
-            Query = query;
+            if (service == WebEnabledFastaImporter.UNIPROTKB_TAG && !string.IsNullOrEmpty(query))
+            {
+                // UniprotKB has gotten tweaky about SGD yeast entries, as of 6/28/2022 only wants to see "S000000001" of "SGD:S000000001"
+                if (query.StartsWith(@"SGD:")) 
+                {
+                    Query = query.Substring(4);
+                }
+                else if (query.StartsWith(@"SGDID:"))
+                {
+                    Query = query.Substring(6);
+                }
+                else
+                {
+                    Query = query;
+                }
+            }
+            else
+            {
+                Query = query;
+            }

```

Comment:
You could flatten/simplify this a bit like:
```
Query = query;
if (service == WebEnabledFastaImporter.UNIPROTKB_TAG)
{
                // UniprotKB has gotten tweaky about SGD yeast entries, as of 6/28/2022 only wants to see "S000000001" of "SGD:S000000001"
                if (query?.StartsWith(@"SGD:")) 
                {
                    Query = query.Substring(4);
                }
                else if (query?.StartsWith(@"SGDID:"))
                {
                    Query = query.Substring(6);
                }
}
Can query actually be null? What would that mean?

Reply:
>> Can query actually be null? 

It only happens during unit tests when we're not actually accessing the web.